src/Cornelis/Goals.hs

```diff
@@ -51,8 +52,8 @@ prevGoal =
   findGoal $ \pos goal ->
     case pos > goal of
       False -> Nothing
-      True -> Just $ ( lineDiff (p_line goal) (p_line pos)
-                     , offsetDiff (p_col goal) (p_col pos)
+      True -> Just $ ( p_line goal .-. p_line pos
+                     , p_col goal .-. p_col pos  -- TODO: This formula looks fishy
```

Comment:
Yeah it does. I think the intent is to get outside of the current goal you might be in, so if your cursor is `{! | !}` you don't jump to where you are.